file/file_prefetch_buffer.cc

```diff
@@ -650,6 +672,14 @@ bool FilePrefetchBuffer::TryReadFromCacheUntracked(
             return false;
           }
         }
+        // Adjust readhahead_size till upper_bound if upper_bound_offset_ is
+        // set.
+        if (upper_bound_offset_ > 0 && upper_bound_offset_ > offset) {
+          if (upper_bound_offset_ < offset + n + readahead_size_) {
+            readahead_size_ = (upper_bound_offset_ - offset) - n;
```

Comment:
Would be good to have a stat for count of readaheads that got trimmed.

Reply:
By stats you mean in statistics.h?

Reply:
Yes. A ticker stat.